src/Akeneo/Pim/Enrichment/Bundle/Controller/ExternalApi/UpdateProductByUuidController.php

```diff
@@ -0,0 +1,338 @@
+        private UrlGeneratorInterface $router,
+        private FilterInterface $emptyValuesFilter,
+        private EventDispatcherInterface $eventDispatcher,
+        private DuplicateValueChecker $duplicateValueChecker,
+        private SecurityFacade $security,
+        private ValidatorInterface $validator,
+        private ObjectUpdaterInterface $updater,
+        private ProductBuilderInterface $productBuilder,
+        private SaverInterface $saver,
+        private AttributeFilterInterface $productAttributeFilter,
+        private Connection $connection,
+        private ValidatorInterface $productValidator,
+    ) {
+    }
+
+    public function __invoke(Request $request, string $uuid): Response
+    {
+        if (!$this->security->isGranted('pim_api_product_edit')) {
+            throw new AccessDeniedHttpException('Access forbidden. You are not allowed to create or update products.');
+        }
+
+        $data = $this->getDecodedContent($request->getContent());
+        $violations = $this->validator->validate($data, new PayloadFormat());
+
+        if (0 < $violations->count()) {
+            $firstViolation = $violations->get(0);
+            $this->throwDocumentedHttpException($firstViolation->getMessage(), new \LogicException($firstViolation->getMessage()));
+        }
+
+        try {
+            $this->duplicateValueChecker->check($data);
+        } catch (InvalidPropertyTypeException $exception) {
+            $this->eventDispatcher->dispatch(new TechnicalErrorEvent($exception));
+            $this->throwDocumentedHttpException($exception->getMessage(), $exception);
+        }
+
+        $product = $this->productRepository->find($uuid);
+
+        $isCreation = false;
+        if (null === $product) {
+            $isCreation = true;
+            if (isset($data['parent'])) {
+                // TODO: fetch identifier attribute code to replace hard coded "sku"
+                $product = $this->productBuilder->createProduct(identifier: $data['values']['sku'][0]['data'] ?? null, uuid: $uuid);
+            } else {
+                $product = $this->productBuilder->createProduct(identifier: null, uuid: $uuid);
+            }
+        }
+
+        $this->validateUuidConsistency($uuid, $data);
+
+        if (!$isCreation) {
+            $data = $this->filterEmptyValues($product, $data);
+        }
+
+        $data = $this->replaceUuidsByIdentifiers($data);
+
+        $this->updateProduct($product, $data);
+        $this->validateProduct($product);
+        $this->saver->save($product);
+
+        return $this->getResponse($product->getUuid(), $isCreation ? Response::HTTP_CREATED : Response::HTTP_NO_CONTENT);
+    }
+
+    private function getDecodedContent($content): array
+    {
+        $decodedContent = json_decode($content, true);
+
+        if (null === $decodedContent) {
+            throw new BadRequestHttpException('Invalid json message received');
+        }
+
+        return $decodedContent;
```

Comment:
Something like this should be better ? (And maybe add more information about the issue).
Can be done in another PR, but let's have a look about what this JsonException gives.

```suggestion
try {
    return json_decode($content, true, 512, JSON_THROW_ON_ERROR);
} catch (JsonException) {
    throw new BadRequestHttpException('Invalid json message received');
}
```

Reply:
Ticket created: https://akeneo.atlassian.net/browse/CPM-718